### PR TITLE
Feature/add some sample data postgres

### PR DESCRIPTION
**Pull request by amorgado**
-  Added the example-data-postgres-book in order to have some data sample application that uses the synapse-data-postgres module and give an example of how to use it (It will be improved in the future with unit tests and other things).
- Added a schema.sql, data.sql, book entity and book repository for the book table to this new module.
- Upgraded H2 and Postgres Jars and tested with the new example-data-postgres-book recently created.

### CODE REVIEW

**Review by jmartinez5120 — CHANGES_REQUESTED** [review]
We need to add Copyrights to all classes. You can copy paste from others.

File: data/example-data-postgres-book/src/main/java/io/americanexpress/data/book/DataBookApplication.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+
+/**
+ * DataBookApplication is the Spring Boot main class to start the data application.
+ */
+@SpringBootApplication
+@EnableConfigurationProperties
```
Comment (jmartinez5120) [inline data/example-data-postgres-book/src/main/java/io/americanexpress/data/book/DataBookApplication.java:11]: I don't think we need `@EnableConfigurationProperties`. I believe `@SpringBootApplication` already has it by default.

File: data/example-data-postgres-book/src/main/java/io/americanexpress/data/book/entity/BookEntity.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+public class BookEntity extends BaseEntity {
+
+    @Column(name = "title")
+    @NaturalId(mutable = true)
+    private String title;
+
+    @Column(name = "author")
+    private String author;
+
+    private String getTitle() {
```
Comment (jmartinez5120) [inline data/example-data-postgres-book/src/main/java/io/americanexpress/data/book/entity/BookEntity.java:25]: `getTitle()` doesn't have JavaDocs

**Review by jmartinez5120 — CHANGES_REQUESTED** [review]
Please check comments.

**Review by cuztomizer — APPROVED** [review]

File: data/example-data-postgres-book/pom.xml [hunk 3]
```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (jmartinez5120) [inline data/example-data-postgres-book/pom.xml:1]: We need to add copyrights in this pom file.

**Review by jmartinez5120 — CHANGES_REQUESTED** [review]
This pom file is missing copyrights. Please add them and we should be good to approve.
`data/example-data-postgres-book/pom.xml`

File: data/example-data-postgres-book/pom.xml (hunk 3 above)
Comment (jmartinez5120) [reply data/example-data-postgres-book/pom.xml:1]: Done

**Review by jmartinez5120 — APPROVED** [review]
LGTM

**Pull request merged**